### DISCUSSION TITLE
Manually setting the position of a body after simulation step, manually changing object layer after simulation step

### DISCUSSION THREAD

**Discussion by captainkidd5** [jrouwe/JoltPhysics#1296] [Q&A] [answered]
I'm trying to add multiple body sensors to my character virtuals, essentially acting as proximity detection (is the enemy within the detection radius of the player? Should items magnetize?).

I need to set the sensors center to be where the player's position is every frame, is there a way to call this?

I also would be interested in changing the body's object layer, say for example if an item on the ground becomes magnetized to the player, I now only want it to collide with the player and not static geometry. Right now I'm just recreating the body, but that feels incorrect

**Comment by jrouwe** [comment, marked as answer]
Use BodyInterface::SetPosition and BodyInterface::SetObjectLayer
